src/map.tsx

```diff
@@ -1,26 +1,143 @@
   control.setProps(props);
   return <></>;
 }
 
-export default function Map({ layers }: { layers: Layer[] }) {
+export default function Map() {
   const mapRef = useRef<MapRef>(null);
   const mapStyle = useColorModeValue(
     "positron-gl-style",
     "dark-matter-gl-style",
   );
+  const [fillColor] = useState<Color>([207, 63, 2, 50]);
+  const [lineColor] = useState<Color>([207, 63, 2, 100]);
+  const [layers, setLayers] = useState<Layer[]>([]);
+  const [valueLayer, setValueLayer] = useState<Layer | null>();
+  const [collectionsLayer, setCollectionsLayer] = useState<Layer | null>();
+  const [stacGeoparquetLayer, setStacGeoparquetLayer] =
+    useState<Layer | null>();
+  const {
+    value,
+    collections,
+    selectedCollections,
+    stacGeoparquetTable,
+    stacGeoparquetMetadata,
+    setStacGeoparquetItemId,
+    stacGeoparquetItem,
+  } = useStacMap();
+
+  useEffect(() => {
+    if (value) {
+      setValueLayer(getValueLayer({ value, lineColor, fillColor }));
+    } else {
+      setValueLayer(null);
+    }
+  }, [value, fillColor, lineColor]);
+
+  useEffect(() => {
+    if (value) {
+      const bbox = getValueBbox(value);
+      if (bbox && mapRef.current) {
+        mapRef.current.fitBounds(sanitizeBbox(bbox), { padding: getPadding() });
+      }
+    }
+  }, [value]);
+
+  useEffect(() => {
+    if (collections) {
+      setCollectionsLayer(
+        getCollectionsLayer({
+          collections,
+          fillColor,
+          lineColor,
+          selectedCollections,
+        }),
+      );
+    } else {
+      setCollectionsLayer(null);
+    }
+  }, [collections, selectedCollections, lineColor, fillColor]);
+
+  useEffect(() => {
+    if (collections) {
+      const bbox = getCollectionsExtent(collections);
+      if (bbox && mapRef.current) {
+        mapRef.current.fitBounds(sanitizeBbox(bbox), { padding: getPadding() });
+      }
+    }
+  }, [collections]);
+
+  useEffect(() => {
+    if (stacGeoparquetTable) {
+      setStacGeoparquetLayer(
+        getStacGeoparquetLayer(
+          stacGeoparquetTable,
+          lineColor,
+          fillColor,
+          setStacGeoparquetItemId,
+          stacGeoparquetItem,
+        ),
+      );
+    } else {
+      setStacGeoparquetLayer(null);
+    }
+  }, [
+    stacGeoparquetTable,
+    fillColor,
+    lineColor,
+    setStacGeoparquetItemId,
+    stacGeoparquetItem,
+  ]);
+
+  useEffect(() => {
+    if (stacGeoparquetMetadata && mapRef.current) {
+      mapRef.current.fitBounds(stacGeoparquetMetadata.bbox, {
+        padding: getPadding(),
+      });
+    }
+  }, [stacGeoparquetMetadata]);
+
+  useEffect(() => {
+    const layers = [];
+    if (stacGeoparquetLayer) {
+      layers.push(stacGeoparquetLayer);
+    }
+    if (collectionsLayer) {
+      layers.push(collectionsLayer);
+    }
+    if (valueLayer) {
+      layers.push(valueLayer);
+    }
+    setLayers(layers);
+  }, [stacGeoparquetLayer, collectionsLayer, valueLayer]);
```

Comment:
Feels much better to be managing the layer stuff here. This will make it easier to customize colors, etc as well b/c stuff won't be scattered all over the place.